app/src/main/java/org/thoughtcrime/securesms/dependencies/UiModule.kt

```diff
@@ -0,0 +1,19 @@
+package org.thoughtcrime.securesms.dependencies
+
+import androidx.recyclerview.widget.RecyclerView
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+class UiModule {
+    @Singleton
```

Comment:
I haven't used Pools before. Is that safe for it to be a singleton instantiated once? This never need to be cleared or anything?

Reply:
I think you are right. Originally I had doubt about the memory leak perspective of using a global pool, at the end of the day, `ViewHolder`s hold `View`s which hold `Context` under the hood. I thought Google should have thought about this and do some magical thing under the hood, until just now you prompted me to have a look again: oh well, nothing magical is happening in the pool so this is def a leaking point. I'm going to remove the use of the pool as they provide marginal benefit anyway. FYI @AL-Session

Reply:
More reasons for us to move this mammoth screen into Compose where we won't have to deal with recyclerviews and viewholders and adapters...